pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -267,6 +267,16 @@ func (r *Revision) GetLastPinned() (time.Time, error) {
 	return time.Unix(secs, 0), nil
 }
 
+// HasRoute returns true if the revision is the target of a route, false otherwise.
+// Determined by the presence of a `serving.RouteLabelKey` label.
+//
+// TODO(tanzeeb) Tests!
+func (r *Revision) HasRoute() bool {
```

Comment:
I'd go with `IsReachable` here and we can tweak the definition over time (e.g. if new routing constructs come to be).